block_iagora.php

```diff
@@ -55,9 +55,49 @@ public function get_content() {
         } else {
             $this->content->text = $this->generate_chat_content($copilotendpointurl);
         }
+
         return $this->content;
 
     }
+    /**
+     * Retrieves the progress of a student in the current course.
```

Comment:
```suggestion
     * Retrieve the progress of a student in the current course.
```
Most functions now use imperative style